cpp/memilio/io/auto_serialize.h

```diff
@@ -0,0 +1,218 @@
+/*
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Rene Schmieding
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef MIO_IO_AUTO_SERIALIZE_H_
+#define MIO_IO_AUTO_SERIALIZE_H_
+
+#include "memilio/io/io.h"
+#include "memilio/utils/metaprogramming.h"
+
+#include <string>
+#include <string_view>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+namespace mio
+{
+
+/**
+ * @brief Pair of name and value used for auto-(de)serialization.
+ *
+ * This object holds a view of a name and reference of a value. Mind their lifetime!
+ * @tparam ValueType The (non-cv, non-reference) type of the value. 
+ */
+template <class ValueType>
+struct NVP {
+    using Type = ValueType&;
+
+    const std::string_view name;
```

Comment:
Hm, apparently string_view does happily accept a string rvalue, and unsurprisingly returns garbage when read from. I thought string_view would be nicer than a char*, but this is definitely an issue.

We can also use string. But before I switched to views that caused several allocations I thought could be easily avoided.

Reply:
Thinking about the allocations. In a call like `Members("Foo").add("i", m_i).add("d", m_d)...`, the copies are not trivial to avoid. `add` would have to be declared like `add(string, ref) &&`, so it can only be called on rvalues and the implementation of add is allowed to move the existing NPVs into the new object it returns. Those are all explicit moves. I don't think the implementation is too difficult. But then most copies/allocations can be avoided and then I would go with the more conventional std::string over const char*

Reply:
I kept char* for now, since it allows the auto_serialize() member functions to be completely inlined. The std::string version moves a quite a bit of data, maybe we could have a look at it together?